### PR TITLE
[settings] Add coded settings button with updateable label

### PR DESCRIPTION
**Pull request by AlwinEsch**
With this it become possible to use a button where the label can continuous updated.

as example:
<code>m_settingsManager->SetString(SETTING_STREAM_INFO_CPU_USAGE, m_CPUUsage); </code>

### CODE REVIEW

File: xbmc/settings/dialogs/GUIDialogSettingsManualBase.cpp [hunk 1]
```diff
@@ -966,6 +984,22 @@ ISettingControl* CGUIDialogSettingsManualBase::GetButtonControl(const std::strin
...
+ISettingControl* CGUIDialogSettingsManualBase::GetInfoButtonControl(const std::string &format, bool delayed /* = false */, int heading /* = -1 */, bool hideValue /* = false */)
+{
+  CSettingControlInfoButton *control = new CSettingControlInfoButton();
+  if (!control->SetFormat(format))
+  {
+    delete control;
+    return NULL;
+  }
+
+  control->SetDelayed(delayed);
```
Comment (AlwinEsch) [inline xbmc/settings/dialogs/GUIDialogSettingsManualBase.cpp]: I can remove this as wanted on other request, but has seen that it is also used on all other types. Please give me a comment.

**Comment by AlwinEsch** [comment]
Can it be reviewed :-) ? Want to update DSP system.

**Comment by AlwinEsch** [comment]
Acceptable?

**Comment by opdenkamp** [comment]
ping @Montellese
jenkins build this please

**Comment by Montellese** [comment]
What exactly is the difference to a normal button? I get that you want to use this for displaying information that constantly changes but I don't get why this can't be done with a normal button. Don't you have to manually change the value of the respective setting in a certain interval or is there some automated logic that is still missing?

**Comment by AlwinEsch** [comment]
On begin it was included to the normal button, but on "infoaction" was not accepted. For me was this before the easier way, there was the button only with another control format.

see:
```C++
+CSettingString* CGUIDialogSettingsManualBase::AddInfoStringButton(CSettingGroup *group, const std::string &id, int label, int level, std::string info, bool delayed /* = false */,
+                                                                  bool visible /* = true */, int help /* = -1 */)
+{
+  if (group == NULL || id.empty() || label < 0 ||
+      GetSetting(id) != NULL)
+    return NULL;
+
+  CSettingString *setting = new CSettingString(id, label, info, m_settingsManager);
+  if (setting == NULL)
+    return NULL;
+
+  setting->SetControl(GetButtonControl("infoaction", delayed));
+  setSettingDetails(setting, level, visible, help);
+
+  group->AddSetting(setting);
+  return setting;
+}
```
and
```C++
@@ -543,7 +544,7 @@ bool CGUIControlButtonSetting::OnClick()
     }
     else if (controlFormat == "path")
       SetValid(GetPath((CSettingPath *)m_pSetting));
-    else if (controlFormat == "action")
+    else if (controlFormat == "action" || controlFormat == "infoaction")
     {
       // simply call the OnSettingAction callback and whoever knows what to
       // do can do so (based on the setting's identification
@@ -614,6 +615,12 @@ void CGUIControlButtonSetting::Update(bool updateDisplayOnly /* = false */)
         if (CUtil::MakeShortenPath(strValue, shortPath, 30))
           strText = shortPath;
       }
+      else if (controlFormat == "infoaction")
+      {
+        strText = strValue;
+        if (strText.empty())
+          strText = g_localizeStrings.Get(231); // None
+      }
     }
   }
   else if (controlType == "slider")
```
if you thing also that this is the better way I correct it back.

**Comment by AlwinEsch** [comment]
Give me a bit and I performing a rework.

**Comment by AlwinEsch** [comment]
Sometimes I think to complicated ;-) (The thinks before), is updated and much more easy and from my side OK. :-)

**Comment by Montellese** [comment]
Hehe, that's what I wanted to point out from the beginning but I probably didn't explain myself very well either.
jenkins build this please

**Comment by AlwinEsch** [comment]
Ready to merge?

File: xbmc/settings/dialogs/GUIDialogSettingsManualBase.cpp [hunk 2]
```diff
@@ -239,6 +239,24 @@ CSettingAction* CGUIDialogSettingsManualBase::AddButton(CSettingGroup *group, co
   return setting;
 }
 
+CSettingString* CGUIDialogSettingsManualBase::AddInfoStringButton(CSettingGroup *group, const std::string &id, int label, int level, std::string info,
```
Comment (Montellese) [inline xbmc/settings/dialogs/GUIDialogSettingsManualBase.cpp]: Please name the method `AddInfoLabelButton()` to match the `infolabel` format name used in the settings library.

**Comment by Montellese** [comment]
Looks good apart from the minor.

**Comment by AlwinEsch** [comment]
Is updated, thanks @Montellese

**Comment by Montellese** [comment]
jenkins build and merge

**Comment by AlwinEsch** [comment]
Thanks :-)

**Pull request merged**